### PR TITLE
New playlists overview implemented

### PR DESCRIPTION
**Pull request by akrol95**

### CODE REVIEW

File: BMM.Core/ViewModels/CuratedPlaylistsViewModel.cs [hunk 1]
```diff
@@ -1,26 +1,61 @@
...
-            return playlists;
+            var playlistsDocuments = await Client
+                .Playlist
+                .GetDocuments(_appLanguageProvider.GetAppLanguage(), _userStorage.GetUser().Age, policy);
+
+            playlistsDocuments
+                .Items
+                .OfType<DiscoverSectionHeader>()
+                .ToList()
+                .ForEach(header => header.UseCoverCarousel = true);
```
Comment (kkuepper) [inline BMM.Core/ViewModels/CuratedPlaylistsViewModel.cs:47]: What's this for? Did I forget this in the API?

File: BMM.Core/ViewModels/CuratedPlaylistsViewModel.cs [hunk 2]
```diff
@@ -1,26 +1,61 @@
...
+            
+            var translatedItems = await _translateDocsAction.ExecuteGuarded(playlistsDocuments.Items.ToList());
+            var carouselAdjustedItems = await _prepareCoversCarouselItemsAction.ExecuteGuarded(translatedItems);
+
+            carouselAdjustedItems
+                .OfType<DiscoverSectionHeader>()
+                .FirstOrDefault()
+                .IfNotNull(header => header.IsSeparatorVisible = false);
```
Comment (kkuepper) [inline BMM.Core/ViewModels/CuratedPlaylistsViewModel.cs:55]: also, why is this needed?

File: BMM.Core/ViewModels/CuratedPlaylistsViewModel.cs [hunk 3]
```diff
@@ -1,26 +1,61 @@
...
 {
     public class CuratedPlaylistsViewModel : DocumentsViewModel
     {
+        private readonly IAppLanguageProvider _appLanguageProvider;
+        private readonly IUserStorage _userStorage;
+        private readonly ITranslateDocsAction _translateDocsAction;
+        private readonly IPrepareCoversCarouselItemsAction _prepareCoversCarouselItemsAction;
         public override CacheKeys? CacheKey => CacheKeys.PlaylistGetAll;
 
         public static readonly List<int> FeaturedIds = new List<int> { 31, 32, 33, 35, 36, 37 };
```
Comment (kkuepper) [inline BMM.Core/ViewModels/CuratedPlaylistsViewModel.cs:23]: this is not related to your PR. But these FeaturedIds is no longer needed. Can you remove it?
(Terrible naming by the way, since it doesn't tell me what it's for.)
As far as I understand, this was needed when we didn't have curated playlists yet. Instead we used a podcast and filtered it out of the list of podcast with these FeaturedIds.

File: BMM.Core/ViewModels/CuratedPlaylistsViewModel.cs (hunk 2 above)
Comment (akrol95) [reply BMM.Core/ViewModels/CuratedPlaylistsViewModel.cs:55]: To avoid showing separator directly under the "Playlists" title, which looks bad:

image

The same mechanism is used on Home and Browse views.

File: BMM.Core/ViewModels/CuratedPlaylistsViewModel.cs (hunk 1 above)
Comment (akrol95) [reply BMM.Core/ViewModels/CuratedPlaylistsViewModel.cs:47]: Yes, but after your changes I'm removing it.

File: BMM.Core/ViewModels/CuratedPlaylistsViewModel.cs (hunk 3 above)
Comment (akrol95) [reply BMM.Core/ViewModels/CuratedPlaylistsViewModel.cs]: Ok, done

**Review by kkuepper — APPROVED** [review]
